pkg/azure/api/api.go

```diff
@@ -386,6 +410,34 @@ func (c *Client) GetInstances(ctx context.Context, subnets ipamTypes.SubnetMap)
 	return instances, nil
 }
 
+var vmssRe = regexp.MustCompile(`[Vv]irtual[Mm]achine[Ss]cale[Ss]ets\/(.*)\/[Vv]irtual[Mm]achines\/(.*)`)
+
+// GetInstance returns the interfaces of a given instance
+func (c *Client) GetInstance(ctx context.Context, subnets ipamTypes.SubnetMap, instanceID string) (*ipamTypes.Instance, error) {
+	instance := ipamTypes.Instance{}
+	instance.Interfaces = map[string]ipamTypes.InterfaceRevision{}
+
+	parts := vmssRe.FindStringSubmatch(instanceID)
+	if len(parts) != 3 {
+		return nil, fmt.Errorf("instance ID %q does not match expected format", instanceID)
```

Comment:
I'm not completely confident that there isn't some dusty edge case in Azure NRP where the instance ID won't match the regex. If that happens, how would this error surface in the logs?

Reply:
I was actually able to replace the regexp with an azure sdk function which should be much more reliable to parse resources IDs.